src/wp-includes/post.php

```diff
@@ -6038,192 +6031,57 @@ function get_pages( $args = array() ) {
-	$last_changed = wp_cache_get_last_changed( 'posts' );
-
-	$cache_key = "get_pages:$key:$last_changed";
-	$cache     = wp_cache_get( $cache_key, 'posts' );
-	if ( false !== $cache ) {
-		_prime_post_caches( $cache, false, false );
-
-		// Convert to WP_Post instances.
-		$pages = array_map( 'get_post', $cache );
-		/** This filter is documented in wp-includes/post.php */
-		$pages = apply_filters( 'get_pages', $pages, $parsed_args );
-
-		return $pages;
-	}
+	$query_args = array(
+		'orderby'                => 'post_title',
+		'order'                  => 'ASC',
+		'post__not_in'           => wp_parse_id_list( $parsed_args['exclude'] ),
+		'meta_key'               => $parsed_args['meta_key'],
+		'meta_value'             => $parsed_args['meta_value'],
+		'author__in'             => wp_parse_list( $parsed_args['authors'] ),
+		'posts_per_page'         => -1,
+		'offset'                 => (int) $parsed_args['offset'],
+		'post_type'              => $parsed_args['post_type'],
+		'post_status'            => $post_status,
+		'update_post_term_cache' => false,
+		'update_post_meta_cache' => false,
+		'ignore_sticky_posts'    => true,
+		'no_found_rows'          => true,
+	);
 
-	$inclusions = '';
 	if ( ! empty( $parsed_args['include'] ) ) {
-		$child_of     = 0; // Ignore child_of, parent, exclude, meta_key, and meta_value params if using include.
-		$parent       = -1;
-		$exclude      = '';
-		$meta_key     = '';
-		$meta_value   = '';
-		$hierarchical = false;
-		$incpages     = wp_parse_id_list( $parsed_args['include'] );
-		if ( ! empty( $incpages ) ) {
-			$inclusions = ' AND ID IN (' . implode( ',', $incpages ) . ')';
-		}
-	}
-
-	$exclusions = '';
-	if ( ! empty( $exclude ) ) {
-		$expages = wp_parse_id_list( $exclude );
-		if ( ! empty( $expages ) ) {
-			$exclusions = ' AND ID NOT IN (' . implode( ',', $expages ) . ')';
-		}
-	}
-
-	$author_query = '';
-	if ( ! empty( $parsed_args['authors'] ) ) {
-		$post_authors = wp_parse_list( $parsed_args['authors'] );
-
-		if ( ! empty( $post_authors ) ) {
-			foreach ( $post_authors as $post_author ) {
-				// Do we have an author id or an author login?
-				if ( 0 == (int) $post_author ) {
-					$post_author = get_user_by( 'login', $post_author );
-					if ( empty( $post_author ) ) {
-						continue;
-					}
-					if ( empty( $post_author->ID ) ) {
-						continue;
-					}
-					$post_author = $post_author->ID;
-				}
-
-				if ( '' === $author_query ) {
-					$author_query = $wpdb->prepare( ' post_author = %d ', $post_author );
-				} else {
-					$author_query .= $wpdb->prepare( ' OR post_author = %d ', $post_author );
-				}
-			}
-			if ( '' !== $author_query ) {
-				$author_query = " AND ($author_query)";
-			}
-		}
-	}
-
-	$join  = '';
-	$where = "$exclusions $inclusions ";
-	if ( '' !== $meta_key || '' !== $meta_value ) {
-		$join = " LEFT JOIN $wpdb->postmeta ON ( $wpdb->posts.ID = $wpdb->postmeta.post_id )";
```

Comment:
what about this post meta query part, isn't that still needed?

Reply:
That is now part of the query args. See 
```
'meta_key'               => $parsed_args['meta_key'],
'meta_value'             => $parsed_args['meta_value'],
```